x-element.d.ts

```diff
@@ -1,69 +1,239 @@
+     * @property {any|Function} [initial]
+     * @property {any|Function} [default]
+     */
+    /**
+     * Declare watched properties (and related attributes) on an element.
+     *
+     *   static get properties() {
+     *     return {
+     *       property1: {
+     *         type: String,
+     *       },
+     *       property2: {
+     *         type: Number,
+     *         input: ['property1'],
+     *         compute: this.computeProperty2,
+     *         reflect: true,
+     *         observe: this.observeProperty2,
+     *         default: 0,
+     *       }
+     *     };
+     *   }
+     * @returns {{[key: string]: Property}} All declared element properties.
+     */
+    static get properties(): {
+        [key: string]: {
+            type?: any;
+            attribute?: string;
+            input?: string[];
+            compute?: Function;
+            observe?: (host: HTMLElement, value: any, oldValue: any) => any;
+            reflect?: boolean;
+            internal?: boolean;
+            readOnly?: boolean;
+            initial?: any | Function;
+            default?: any | Function;
+        };
+    };
+    /**
+     * Listen callback.
+     * @callback delegatedListenCallback
+     * @param {HTMLElement} host
+     * @param {Event} event
+     */
+    /**
+     * Declare event handlers on an element.
+     *
+     *   static get listeners() {
+     *     return {
+     *       click: this.onClick,
+     *     }
+     *   }
+     *
+     * Note that listeners are added to the element's render root. Listeners are
+     * added during "connectedCallback" and removed during "disconnectedCallback".
+     * The arguments passed to your callback are always "(host, event)".
+     * @returns {{[key: string]: delegatedListenCallback}} All element listeners.
+     */
+    static get listeners(): {
+        [key: string]: (host: HTMLElement, event: Event) => any;
+    };
+    /**
+     * Customize shadow root initialization and optionally forgo encapsulation.
+     * E.g., setup focus delegation or return host instead of host.shadowRoot.
+     * @param {HTMLElement} host
+     * @returns {HTMLElement|ShadowRoot} a
+     */
+    static createRenderRoot(host: HTMLElement): HTMLElement | ShadowRoot;
+    /**
+     * Template callback.
+     * @callback templateCallback
+     * @param {object} properties
+     * @param {HTMLElement} host
+     */
+    /**
+     * Setup template callback to update DOM when properties change.
+     *
+     *   static template(html, { nullish }) {
+     *     return (href) => {
+     *       return html`<a href=${nullish(href)}>click me</a>`;
+     *     }
+     *   }
+     * @param {Function} html
+     * @param {{[key: string]: Function}} engine
+     * @returns {templateCallback}
+     */
+    static template(html: Function, engine: {
+        [key: string]: Function;
+    }): (properties: object, host: HTMLElement) => any;
+    static "__#1@#analyzeConstructor"(constructor: any): void;
```

Comment:
… This, is annoying. TypeScript does want to track private fields for [reasons], so as far as I can tell, you cannot remove this stuff.